src/main/java/com/google/cloud/genomics/utils/grpc/SequenceUtils.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.genomics.utils.grpc;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.Predicate;
+
+import java.util.List;
+
+/**
+ * Various sequence and genome utility functions
+ */
+public class SequenceUtils {
+
+  /** Types of genotypes possible for diploid and haploid variants based on zygosity. */
+  public enum GenotypeCategory {
+    REF_HOMOZYGOUS, // Diploid
+    HETEROZYGOUS, // Diploid
+    NON_REF_HOMOZYGOUS, // Diploid
+    REF, // Haploid
+    NON_REF // Haploid
+  }
+
+  /**
+   * Classifies a genotype based on zygosity and ploidy.
+   *
+   * @param genotype list of integers representing a genotype within which 0 represents the
+   *    reference allele and values greater than 0 represent alternatives.
+   * @return Classification of this genotype base on zygosity, with special handling for haploids.
+   */
+  public static final GenotypeCategory classifyGenotype(List<Integer> genotype) {
+    boolean isHaploid = genotype.size() < 2;
+    boolean hasRef = false;
+    boolean allRef = true;
+    for (Integer allele : genotype) {
+      if (allele == 0) {
+        hasRef = true;
+      } else {
+        allRef = false;
+      }
+    }
+
+    if (allRef) {
+      // Reference homozygous iff {0,0}, haploid reference if {0}
```

Comment:
s/iff/if and only if

Most folks might know know the acronym, so to be on the safe side it is best to expand all acronyms unless they are obvious such as DNA, etc.